File: week2-arrays/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    //prompt user for the text
    string text = get_string("Text: ");

    //number of letters in the text
    int letters = 0;
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
            letters++;
    }

    // number of words in text
    int words = 1;
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if(text[i] == ' ')
            words++;
    }

    //number of sentences in text
    int sentences = 0;
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if(text[i] == '.' || text[i] == '!' || text[i] == '?')
            sentences++;
    }

    //Coleman-Lian index
    float calculation = (0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8;
    int index = round(calculation);

    if(index < 1)
        printf("Before Grade 1\n");
    else if(index >= 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", index);
}

File: week6-python/pset6/weeks/dna/sentimental-readability/readability.py
#prompt text from user
text = input("Text: ")

letters = 0
words = 1
sentences = 0

for i in text:
    if i.isalpha():
        letters+=1
    elif i == " ":
        words+=1
    elif i in [".", "!", "?"]:
        sentences+=1

l = (letters / words) * 100
s = (sentences / words) * 100
index = 0.0588 * l - 0.296 * s - 15.8

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print ("Grade 16+")
else:
    print("Grade ", round(index))